capstone/files/tests/test_main.py

```diff
@@ -0,0 +1,65 @@
+from fastapi.testclient import TestClient
+from fastapi import UploadFile
+
+from os import path
+
+import builtins
+import main
+import tempfile
+import pytest
+
+client = TestClient(main.app)
+
+
+def test_homepage():
+    response = client.get('/')
+    assert response.status_code == 200
+    assert response.template.name == "form.html.jinja"
+    assert "request" in response.context
+
+
+def test_download_file_ok(monkeypatch):
+    f = tempfile.NamedTemporaryFile()
```

Comment:
No need to use a tempfile, you can mock "open" to return what you expect.

Reply:
`download_file` is doing a few things.
1. it's joining paths
1. it's checking if path is an actual file
1. it returns a `FileResponse`

If don't even use `open` here so I don't quite get what should I mock that will be better then `join`.
Right now I can mock only one thing (`os.path.join`) and the whole function is working correctly.

Reply:
In this case it makes sense to use a tempfile, you're right :)